AnimView/th.cpp

```diff
@@ -179,10 +179,14 @@ void decode_chunks_complex(ChunkRenderer& renderer, const uint8_t* data,
   renderer.chunkFinish(transparent);
 }
 
+inline uint8_t average_colour(unsigned int channel_a, unsigned int channel_b) {
+    return (channel_a + channel_b) / 2;
```

Comment:
This is a narrowing operation, let's cast.

Reply:
Arguably the function could be
``` cpp
inline uint8_t average_colour(uint8_t channel_a, uint8_t channel_b) {
    unsigned int ch_a = channel_a;
    unsigned int ch_c = channel_b;
    return (ch_a + chl_b) / 2;
}
```
to clarify that channels are 8 bit, but I fail to understand why casting the return expression does anything. Clearly the return type says we return 8 bits only?

Reply:
because implicit conversion is a warning, and narrowing conversions in particular should be pointed out explicitly because they can overflow (though this one can't). I do prefer that signature.